### PR TITLE
Rdm 2226 DB layer: Add new OrderSummary db entry in field_type

### PR DESCRIPTION
**Pull request by andrewfolga**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/RDM-2226

Change description
DB layer: Add new OrderSummary db entry in field_type + validation

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x] No

### CODE REVIEW

**Comment by tzarsmango** [comment]
Reviewed after this [commit](https://github.com/hmcts/ccd-definition-store-api/pull/34/commits/a993b139da994159aa7044301572c56c25f22e41l) Everything seems fine 👍

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/OrderSummaryFieldTypeToDisplayContextValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        assertAll(
+            () -> assertThat(validationError.isPresent(), is(true)),
+            () -> assertThat(validationError.get(),
+                             hasProperty("defaultMessage", equalTo("OrderSummary field type can only be configured with 'READONLY' DisplayContext")))
+        );
+    }
+
+    @Test
+    public void shouldReturnNullIfFieldTypeReadonly() throws Exception {
```
Comment (andrewfolga) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/OrderSummaryFieldTypeToDisplayContextValidatorTest.java]: fix test name

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/OrderSummaryFieldTypeToDisplayContextValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        EventCaseFieldEntity eventCaseFieldEntity = new EventCaseFieldEntity();
+        eventCaseFieldEntity.setDisplayContext(DisplayContext.READONLY);
+        CaseFieldEntity caseFieldEntity = new CaseFieldEntity();
+        FieldTypeEntity fieldType = new FieldTypeEntity();
+        fieldType.setReference(BASE_ORDER_SUMMARY);
+        caseFieldEntity.setFieldType(fieldType);
+        eventCaseFieldEntity.setCaseField(caseFieldEntity);
+
```
Comment (andrewfolga) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/OrderSummaryFieldTypeToDisplayContextValidatorTest.java]: Add a test for missing value.

Comment (andrewfolga) [reply domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/OrderSummaryFieldTypeToDisplayContextValidatorTest.java]: This is handled one level higher up in EventCaseFieldDisplayContextValidatorImpl. A relevant test has been updated there.

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/OrderSummaryFieldTypeToDisplayContextValidatorTest.java (hunk 1 above)
Comment (andrewfolga) [reply domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefield/OrderSummaryFieldTypeToDisplayContextValidatorTest.java]: Done

**Pull request merged**